File: Baekjoon/누적합/11411.py
# https://www.acmicpc.net/problem/11441

import sys
input = sys.stdin.readline

n = int(input())
nums = list(map(int, input().split()))
sums = [0] * (n + 1)

for i in range(1, n + 1):
    sums[i] = sums[i - 1] + nums[i - 1]

m = int(input())

for _ in range(m):
    i, j = map(int, input().split())
    print(sums[j] - sums[i - 1])

File: Baekjoon/누적합/11411.cpp
// https://www.acmicpc.net/problem/11441

#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;
    cin >> n;
    vector<int> arr(n + 1);
    for (int i = 1; i <= n; i++) {
        cin >> arr[i];
        arr[i] += arr[i - 1];
    }

    int m;
    cin >> m;
    while (m--) {
        int a, b;
        cin >> a >> b;
        cout << arr[b] - arr[a - 1] << '\n';
    }

    return 0;
}